frontend/src/components/common/AdminUserTable.tsx

```diff
@@ -19,6 +19,18 @@ interface AdminUserTableProps {
 const AdminUserTable = ({
   adminUsers,
 }: AdminUserTableProps): React.ReactElement => {
+  const [users, setUsers] = useState(adminUsers);
+  // const [order, setOrder] = useState("Ascending");
+  // const sorting = (property: string)=>{
+  //   if (order == "Ascending") {
+  //     const sorted = [...users].sort((a, b)=>
+  //     a[property].toLowerCase() > b[property].toLowerCase() ? 1 : -1
+  //     );
+  //     setUsers(sorted);
+  //     setOrder("Descending")
+  //   };
+  // };
```

Comment:
```suggestion
type AdminUserProperty = "firstName" | "lastName" | "email";

const AdminUserTable = ({
  adminUsers,
}: AdminUserTableProps): React.ReactElement => {
  const [users, setUsers] = useState(adminUsers);
  const [order, setOrder] = useState("Ascending");
  const sorting = (property: AdminUserProperty) => {
     if (order === "Ascending") {
       setUsers(users.sort((a, b)=>
         a[property].toLowerCase() > b[property].toLowerCase() ? 1 : -1
       ));
       setUsers(sorted);
       setOrder("Descending")
     }
   };
```
This should fix your userTypes parsing issue! TypeScript was complaining because objects of type `AdminUser` can only have the props `firstName`, `lastName`, or `email`, which is more specific than just a general string type

Reply:
Thank you so much!!! That had me stuck for a while LOL